subprojects/GOPATH/src/github.com/microsoft/hivedscheduler/pkg/algorithm/cell.go

```diff
@@ -121,30 +118,58 @@ func (c *GenericCell) IncreaseUsedGpuNumAtPriority(p CellPriority, delta int32)
 // PhysicalCell defines a cell in the physical cluster.
 type PhysicalCell struct {
 	GenericCell
-	nodes               []string           // node names inside the cell
-	gpuIndices          []int32            // [-1] for cells at levels higher than node
-	affinityGroup       *AlgoAffinityGroup // affinity group using this cell
-	virtualCell         *VirtualCell       // points to the bound virtual cell
-	preBoundVirtualCell *VirtualCell       // points to the temporarily bound virtual cell (before the binding is confirmed)
-	split               bool               // true when the cell has been split
-	reserved            bool               // true when this is a reserved cell
+	nodes               []string                // node names inside the cell
+	gpuIndices          []int32                 // [-1] for cells at levels higher than node
+	affinityGroup       *AlgoAffinityGroup      // affinity group using this cell
+	virtualCell         *VirtualCell            // points to the bound virtual cell
+	preBoundVirtualCell *VirtualCell            // points to the temporarily bound virtual cell (before the binding is confirmed)
+	split               bool                    // true when the cell has been split
+	reserved            bool                    // true when this is a reserved cell
+	status              *api.PhysicalCellStatus // json representation of the status of this cell to expose to users
 }
 
-func NewPhysicalCell(c CellChain, l CellLevel, g bool, n int32) *PhysicalCell {
+func NewPhysicalCell(c CellChain, l CellLevel, g bool, n int32, cellType string, address string) *PhysicalCell {
 	return &PhysicalCell{
 		GenericCell: GenericCell{
 			chain:                  c,
 			level:                  l,
 			priority:               freePriority,
+			address:                address,
 			atOrHigherThanNode:     g,
 			totalGpuNum:            n,
 			usedGpuNumAtPriorities: map[CellPriority]int32{},
 		},
+		status: &api.PhysicalCellStatus{
+			CellStatus: api.CellStatus{
+				CellType:    cellType,
+				CellAddress: address,
+				State:       api.FreeState,
+				Priority:    int32(freePriority),
+			},
+		},
 	}
 }
 
-func (c *PhysicalCell) GetName() string {
-	return fmt.Sprintf("physical.%v.L%d.%v", c.chain, c.level, c.GetPhysicalPlacementString())
+func (c *PhysicalCell) SetPriority(p CellPriority) {
+	c.priority = p
+	c.status.Priority = int32(p)
+	state := api.UsedState
+	if p == freePriority {
+		state = api.FreeState
+	}
+	c.status.State = state
+	if c.status.VirtualCell != nil {
```

Comment:
If the old state is bad, will it override it to be others?

Reply:
Solved as we have used separated states for bad and used